src/main/java/com/epam/reportportal/karate/KarateReportPortalSuite.java

```diff
@@ -0,0 +1,378 @@
+package com.epam.reportportal.karate;
+
+import com.intuit.karate.*;
+import com.intuit.karate.core.*;
+import com.intuit.karate.driver.DriverRunner;
+import com.intuit.karate.http.HttpClientFactory;
+import com.intuit.karate.job.JobManager;
+import com.intuit.karate.report.ReportUtils;
+import com.intuit.karate.report.SuiteReports;
+import com.intuit.karate.resource.Resource;
+import com.intuit.karate.resource.ResourceUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Stream;
+
+public class KarateReportPortalSuite extends Suite {
+    private static final Logger logger = LoggerFactory.getLogger(Suite.class);
+
+    public final long startTime;
+    protected long endTime;
+    protected int skippedCount;
+    private final AtomicBoolean abort;
+    public final String env;
+    public final String tagSelector;
+    public final boolean dryRun;
+    public final boolean debugMode;
+    public final File workingDir;
+    public final String buildDir;
+    public final String reportDir;
+    public final ClassLoader classLoader;
+    public final int threadCount;
+    public final int timeoutMinutes;
+    public final int featuresFound;
+    public final List<FeatureCall> features;
+    public final List<CompletableFuture<Boolean>> futures;
+    public final Set<File> featureResultFiles;
+    public final Collection<RuntimeHook> hooks;
+    public final HttpClientFactory clientFactory;
+    public final Map<String, String> systemProperties;
+    public final boolean backupReportDir;
+    public final SuiteReports suiteReports;
+    public final boolean outputHtmlReport;
+    public final boolean outputCucumberJson;
+    public final boolean outputJunitXml;
+    public final boolean parallel;
+    public final ExecutorService scenarioExecutor;
+    public final ExecutorService pendingTasks;
+    public final JobManager<KarateReportPortalRunner> jobManager;
+    public final String karateBase;
+    public final String karateConfig;
+    public final String karateConfigEnv;
+    public final Map<String, Object> callSingleCache;
+    public final Map<String, ScenarioCall.Result> callOnceCache;
```

Comment:
What is the reason of making these fields public?

Reply:
Deleted whole KarateReportPortalSuite.java class